app/models/ems_event.rb

```diff
@@ -76,10 +76,6 @@ def self.add_openstack_infra(ems_id, event)
     add(ems_id, ManageIQ::Providers::Openstack::InfraManager::EventParser.event_to_hash(event, ems_id))
   end
 
-  def self.add_amazon(ems_id, event)
-    add(ems_id, ManageIQ::Providers::Amazon::CloudManager::EventParser.event_to_hash(event, ems_id))
-  end
```

Comment:
FWIW, we could:

```ruby
def self.add_generic(ems_id, event)
  ems = ExtManagementSystem.find(ems_id)
  add(ems_id, ems.class::EventParser.event_to_hash(event, ems_id))
end
```

(ignoring any further potential to pass `ems` in place of `ems_id` and avoid a later double-lookup)